### DISCUSSION TITLE
akka-http route attribute

### DISCUSSION THREAD

**Discussion by imaxmelnyk** [open-telemetry/opentelemetry-java-instrumentation#7691] [Q&A]
Hi guys,

For the `akka-http` instrumentaiton, `route` attribute, of http server metrics, is not exported due to hardly achievable low cardinality of it, that is fine and i understand it. However, is there any way to fill it via either extension or manual instrumentation?
Say, if i put the route attribute inside the `HttpRequest` object so that i reuse it in the extension later on to fill `route`? is it even possible?

Thanks

**Comment by mateuszrzeszutek** [comment]
Hey @imaxmelnyk ,

If you have access to the route in your application you can use the [`HttpRouteHolder`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpRouteHolder.java) class to pass it to the instrumentation while the server span is still not ended.
Note that this is an unstable API, and there might be some changes made to it in the future; we're currently working on stabilizing the HTTP conventions spec, so it may take several months until this gets a stable release.
